### PR TITLE
feat: tier with protected ingress

### PR DESCRIPTION
**Pull request by gordalina**
## Summary

Finished implementation for tier.

A few questions:

1. Is there an example dashboard that I can re-use for tier?
2. I'm not sure I've addressed the secrets correctly, both the stripe API key that needs to be input as well as the basic auth key that's generated upon install

## Test Plan

1. Create a live and test key in stripe
3. Configure tier with live=false and use the live stripe key
4. Configure tier with live=true and use the test stripe key
5. If the pod runs, it means it can talk to stripe correctly. Alternatively you can run `curl -u "{{.Values.tier.basicAuth }}:" https://{{ .Values.hostname }}`

## Checklist

- [x]  No images hosted from dockerhub
- [ ]  Are dashboards present to understand the health of the application.  There **must** be at least 1 of these
    - [ ]  all databases should have dashboards
    - [ ]  ideally also have at least cpu/mem utilization dashboards for webserver tier of the app
    - [ ]  you can use `plural from-grafana` to convert a grafana dashboard found via google to our CRD
- [ ]  Are scaling runbooks present
    - [ ]  all databases **must** have scaling runbooks
    - [ ]  you can use the charts in `pluralsh/module-library` to accelerate this
- [ ]  do you need to add config overlays?
    - [ ]  inputing secrets
    - [ ]  configuring autoscaling
- [ ]  If there’s a web-facing component to the app, we need to support OIDC authentication and setting up private networks if no authentication option is viable
- [x]  All major clouds must be supported
    - [x]  Azure
    - [x]  AWS
    - [x]  GCP

### CODE REVIEW

File: tier/helm/tier/templates/deployment.yaml [hunk 1]
```diff
@@ -33,17 +33,26 @@ spec:
...
+                name: {{ include "tier.fullname" . }}-env
           ports:
             - name: http
-              containerPort: 80
+              containerPort: 8080
               protocol: TCP
           livenessProbe:
             httpGet:
-              path: /
+              path: /v1/whoami
```
Comment (michaeljguarino) [inline tier/helm/tier/templates/deployment.yaml:51]: is this a suitable healthcheck?  Not sure if it ends up being an expensive op or not

**Review by michaeljguarino — APPROVED** [review]
This should be good to go actually.  The stripe key will be stored securely since the values files are encrypted in git, so no worry there.

As far as dashboards/runbooks, my guess is the only thing needed for an app this thin would be a minor scaling runbook that allows you to modify the replicas for the app and tune its cpu/memory requests.  You could probably copypasta strapi's runbook for for that

File: tier/plural/notes.tpl [hunk 2]
```diff
@@ -1 +1,5 @@
 Your tier installation is available at https://{{ .Values.hostname }}
+
+You can inspect the runtime configuration by running:
+
+  curl -u "{{ .Values.tier.basicAuth }}:" https://{{ .Values.hostname }}/v1/whoami
```
Comment (michaeljguarino) [inline tier/plural/notes.tpl:5]: isn't the basic auth here: `-u "auth:{{ .Values.tier.basicAuth }}"` or am I misunderstanding

File: tier/helm/tier/templates/deployment.yaml (hunk 1 above)
Comment (gordalina) [reply tier/helm/tier/templates/deployment.yaml:51]: This endpoint does talk to the Stripe API, but it is a true liveness check. If it doesn't work its means its improperly configured.

File: tier/plural/notes.tpl (hunk 2 above)
Comment (gordalina) [reply tier/plural/notes.tpl:5]: In your example, `auth` is the user and `basicAuth` would be the password, but I'm using `basicAuth` as the user without a password. It's a common use case when you only want to simulate an API key.

**Comment by gordalina** [comment]
@michaeljguarino added the scaling runbook.

**Review by michaeljguarino — APPROVED** [review]
Awesome, thanks for the contribution!

**Pull request merged**